gb-cpu/src/microcode/bitwise.rs

```diff
@@ -89,3 +89,31 @@ pub fn rr(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
     state.regs.set_subtraction(false);
     OK_PLAY_NEXT_ACTION
 }
+
+pub fn sla(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
+    let mut value = ctl.pop();
+    let top_bit = value >> 7;
+    value <<= 1;
+
+    ctl.push(value);
+
+    state.regs.set_zero(value == 0);
+    state.regs.set_carry(top_bit == 1);
+    state.regs.set_half_carry(false);
+    state.regs.set_subtraction(false);
+    OK_PLAY_NEXT_ACTION
+}
+
+pub fn sra(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
```

Comment:
from the documentation of `SRA` the 7th bit remain unchanged, here it will be shifted to the right